core/application/src/app.rs

```diff
@@ -15,7 +16,7 @@ use crate::config::{Config, StorageConfig};
 use crate::env::{Env, UpdateWorker};
 use crate::query_runner::QueryRunner;
 pub struct Application<C: Collection> {
-    update_socket: ExecutionEngineSocket,
+    update_socket: Arc<Mutex<Option<ExecutionEngineSocket>>>,
```

Comment:
```suggestion
    update_socket: Mutex<Option<ExecutionEngineSocket>>,
```
No need for the Arc here